### PR TITLE
Update the grid theme

### LINKED ISSUE #8833: Themes: Grid selected cell background not listed in theme props
## Description  
The `igx-grid-theme` should list a property for `$row-selected-cell-background` instead of trying to calculate it from the row-selected-background as it currently is. The property should be added to both the theme and the schemas of the grid.

 * igniteui-angular version: all
 * browser: all

### PR DESCRIPTION
**Pull request by desig9stein**
Closes #8833 

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_grid.scss [hunk 1]
```diff
@@ -197,10 +205,19 @@ $_light-grid: extend(
...
         ),
 
+        row-col-selected-cell-background: (
+            igx-color: ('secondary', 50)
+        ),
+
+        row-col-selected-cell-color: (
+            igx-contrast-color: ('secondary', 50)
+        ),
+
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_grid.scss:220]: This always gets overridden by the code executed on lines 258 through 264 in `_grid-theme.scss`;

**Review by simeonoff — CHANGES_REQUESTED** [review]
Since you add explicit schema properties for setting cell selected text/background colors when in a selected row/column, you don't need to calculate the color based off of the luminance of the parent selected row. You can define these colors inside the grid schema for each theme.

File: projects/igniteui-angular/src/lib/core/styles/themes/schemas/dark/_grid.scss [hunk 2]
```diff
@@ -74,13 +78,28 @@ $_base-dark-grid: (
         hexrgba: #222
     ),
 
+    cell-selected-within-background: (
+        igx-color: ('secondary', 500),
+        rgba: .24,
+        hexrgba: #222,
+        lighten: 8%
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/schemas/dark/_grid.scss:85]: I really hate that we have to do all this :( Isn't there a color from the secondary palette that is close to the color produced from all this?

**Review by simeonoff — CHANGES_REQUESTED** [review]
It would be awesome if we could use colors from the palette directly, instead of having to produce exclusive colors...

File: projects/igniteui-angular/src/lib/core/styles/themes/schemas/dark/_grid.scss (hunk 2 above)
Comment (desig9stein) [reply projects/igniteui-angular/src/lib/core/styles/themes/schemas/dark/_grid.scss:85]: This is the closest I can get  [screenshot 1](https://i.imgur.com/iBuWFwx.png), [screenshot 2](https://i.imgur.com/w3ey2DX.png)

Comment (simeonoff) [reply projects/igniteui-angular/src/lib/core/styles/themes/schemas/dark/_grid.scss:85]: The results in screenshot 1 look decent.

**Review by simeonoff — CHANGES_REQUESTED** [review]
The selection colors in the light fluent themes looks bad :(

**Review by simeonoff — APPROVED** [review]

**Pull request merged**